r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+
+- `"Style selection"`: Highlight text and click this button to reformat the code.
+- `"Style active file"`: Code in the `.R` or `.Rmd` file you have open and
+visible in RStudio will be reformatted when you click this button.
+
+There are two other styler functions that are also useful:
+
+- `styler::style_file("path/to/filename")`: Styles the whole file as indicated by the
+file path in the first argument. Can be either an `.R` or `.Rmd` file.
+- `styler::style_dir("directoryname")`: Styles all files in the indicated directory
+in the first argument.
+
+Let's try the `styler::style_file()` function out. Inside a file called
+`non-styled-code.R`, it has:
+
+```{r r-repro-non-styled-code, eval=FALSE}
+# Spacing
+x[,1]
+mean (x, na.rm = TRUE)
+function (x) {}
+height<-feet*12+inches
+sqrt(x ^ 2 + y ^ 2)
+df $ z
+x <- 1 : 10
+
+# Indenting
+if (y < 0 && debug)
+message("Y is negative")
+```
+
+Then we run:
+
+```{r r-repro-use-styler-file, eval=FALSE}
+styler::style_file("testing-styler.R")
+```
+
+```
+Styling  1  files:
+ testing-styler.R ℹ 
+────────────────────────────────────────
+Status	Count	Legend 
+✔ 	0	File unchanged.
+ℹ 	1	File changed.
+✖ 	0	Styling threw an error.
+────────────────────────────────────────
+Please review the changes carefully!
+```
+
+Which changes the file to be styled!
+
+```{r r-repro-styled-code, eval=FALSE}
+# Spacing
+x[, 1]
+mean(x, na.rm = TRUE)
+function(x) {}
+height <- feet * 12 + inches
+sqrt(x^2 + y^2)
+df$ z
+x <- 1:10
+
+# Indenting
+if (y < 0 && debug) {
+  message("Y is negative")
+}
+```
+
+This is more or less everything that the styler package does.
+
+### Exercise: Use styler to fix code formatting
+
+Use the styler package function on the code from the previous exercise by either
+running `styler::style_file()` or with the `"Style selection"` addin when
+highlighting the code.
+
+## Integrating text, code, and results
+
+The most obvious demonstration of reproducibility is when the results obtained
+from executing the analysis code (by an independent entity) are indentical to
+the results presented in the scientific output such as in an article. When there
+is agreement between these two, reproducibility has been more or less acheived.
+In R, there are tools available to *completely* ensure that this happens by
+directly inserting the results from the code *into the scientific ouput*. This is
+done by using [R Markdown][r-markdown], which interweaves R code with text.
+So instead of, for example, manually inserting a figure, you write R code within
+the document to insert the figure for you! Using [R Markdown][r-markdown] can
+save so much time and get your work that much closer to being reproducible.
+
+There are many other advantages to using R Markdown. From the single R Markdown
+format you can use it to create manuscripts, posters, slides, websites, books,
+and many more from simply using R Markdown! In fact, this book was written using
```

Comment:
Is! This! Really! Exciting! Enough! To! Merit! An! Exclamation! Mark!??

Reply:
Hahah, sometimes my enthusiasm gets the better of me and I just! NEED! TO! USE! THEM! :wink: I get your point.

Reply:
And yes, I do think it is pretty exciting haha